torch/csrc/jit/passes/fuse_relu.cpp

```diff
@@ -0,0 +1,94 @@
+
+#include <torch/csrc/jit/passes/fuse_relu.h>
+#include <torch/csrc/jit/ir/ir.h>
+#include <torch/csrc/jit/ir/subgraph_matcher.h>
+#include <torch/csrc/jit/passes/subgraph_rewrite.h>
+
+namespace torch {
+namespace jit {
+
+namespace {
+void fuseAddReluImpl(std::shared_ptr<Graph>& graph) {
+  SubgraphRewriter rewriter;
+
+  std::string add_relu_0 = R"(
+    graph(%a, %b, %alpha):
+        %add_res = aten::add(%a, %b, %alpha)
+        %res = aten::relu(%add_res)
+        return (%res))";
+  std::string add_relu_fused_0 = R"(
+    graph(%a, %b, %alpha):
+        %res = aten::add_relu(%a, %b, %alpha)
+        return (%res))";
+  rewriter.RegisterRewritePattern(add_relu_0, add_relu_fused_0);
+
+  std::string add_relu_1 = R"(
+    graph(%a, %b, %alpha):
+        %add_res = aten::add(%a, %b, %alpha)
+        %res = aten::relu_(%add_res)
+        return (%res))";
+  // If add was followed by in place relu then
+  // we should probably make fused add_relu also
+  // inplace. since in the graph, it does not seem
+  // that output of add will be really used anywhere else.
```

Comment:
I don't agree with this, since you're now mutating one of the addends, rather than the sum.

Reply:
Hmm. I think you are right. I need to check a graph like this:
```
def m(a, b, c):
   tmp = a.add(b)
   d = a.add(c)
   x = self.relu_(tmp)
   return x + d
```
If add + relu_ is replaced with add_relu_ then a is mutated and if a.add(c) is executed afterwards that is wrong. This transformation adds a new alias. In the original graph `tmp = a.add(b)` a and tmp dont' alias but replacing that with `tmp = a.add_relu_(b)` now I have made a and tmp alias each other.
I will test and fix this.

Reply:
@dreiss I fixed this and also added exactly this test that I mentioned above. Without fixing it, the test fails, after the fix it passes.